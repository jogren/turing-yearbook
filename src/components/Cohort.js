import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = ({people, deleteMember}) => {
  let staffList = people.map(person => {
    return <Person id={person.id} key={person.id} name={person.name} quote={person.quote} superlative={person.superlative} photo={person.photo} deleteMember={deleteMember} />
  })
  
  return (
    <main className="Cohort-main">
      {staffList}
    </main>
  )
}

export default Cohort;
