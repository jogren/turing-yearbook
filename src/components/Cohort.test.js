import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Cohort 
      people={ [{id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'}] }
      deleteMember={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
})