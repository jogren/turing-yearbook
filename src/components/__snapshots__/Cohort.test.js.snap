// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all data passed in correctly 1`] = `
<main
  className="Cohort-main"
>
  <Person
    deleteMember={[MockFunction]}
    id={1}
    key="1"
    name="Eric"
    photo="./eric.jpg"
    quote="You miss all the shots you don't take."
    superlative="Most Likely to Win the PGA"
  />
</main>
`;
